sew.lic

```diff
@@ -54,16 +48,18 @@ class Sew
     @stamp = @settings.mark_crafted_goods
     @training_spells = @settings.crafting_training_spells
 
-    wait_for_script_to_complete('buff', ['sew'])
+    DRC.wait_for_script_to_complete('buff', ['sew'])
 
     Flags.add('sew-assembly', 'another finished \S+ shield (handle)', 'another finished wooden (hilt|haft)', 'another finished (long|short|small|large) leather (cord|backing)', 'another finished (small|large) cloth (padding)', 'another finished (long|short) wooden (pole)')
     Flags.add('sew-pins', 'The pins is all used up, so you toss it away.')
 
     if args.type == 'knitting'
       knit
     elsif args.type == 'sewing'
+      @mat_noun = 'cloth'
```

Comment:
I see several references to `cloth` and `leather` still in the script. And the single use of `@mat_noun` is in a common helper whose calling function already knows if it is cloth or leather.

What do you think about adding a `material_type` argument to `do_sew`?

Reply:
So far as i could tell, there was only one spot i missed. One reason NOT to add that argument is that as soon as you cut it with scissors, it's no longer "cloth" or "leather", but is now "cloak" or "gloves". submitting the change.